### PR TITLE
Update mem limit for MPS

### PR DESCRIPTION
**Pull request by grac3gao**

### CODE REVIEW

File: pkg/gpu/nvidia/manager_test.go [hunk 1]
```diff
@@ -169,7 +169,7 @@ func Test_nvidiaGPUManager_Envs(t *testing.T) {
 		},
 		{
 			name:           "MPS enabled, single GPU request",
-			totalMemPerGPU: 80 * 1024,
+			totalMemPerGPU: 80 * 1024 * 1024 * 1024,
```
Comment (richardsliu) [inline pkg/gpu/nvidia/manager_test.go:172]: Add a comment for the total limit (80GB)?

File: pkg/gpu/nvidia/manager_test.go [hunk 2]
```diff
@@ -179,12 +179,12 @@ func Test_nvidiaGPUManager_Envs(t *testing.T) {
 			numDevicesRequested: 1,
 			want: map[string]string{
 				mpsThreadLimitEnv: "10",
-				mpsMemLimitEnv:    "8192MB",
+				mpsMemLimitEnv:    "0=8192M",
```
Comment (richardsliu) [inline pkg/gpu/nvidia/manager_test.go:182]: Add an explanation for what "0=" means?

File: pkg/gpu/nvidia/manager.go [hunk 3]
```diff
@@ -313,12 +313,10 @@ func (ngm *nvidiaGPUManager) Envs(numDevicesRequested int) map[string]string {
 	if ngm.gpuConfig.GPUSharingConfig.GPUSharingStrategy == gpusharing.MPS {
 		activeThreadLimit := numDevicesRequested * 100 / ngm.gpuConfig.GPUSharingConfig.MaxSharedClientsPerGPU
 		memoryLimit := uint64(numDevicesRequested) * ngm.totalMemPerGPU / uint64(ngm.gpuConfig.GPUSharingConfig.MaxSharedClientsPerGPU)
-
 		return map[string]string{
 			mpsThreadLimitEnv: strconv.Itoa(activeThreadLimit),
-			mpsMemLimitEnv:    fmt.Sprintf("%dMB", memoryLimit),
+			mpsMemLimitEnv:    fmt.Sprintf("0=%dM", memoryLimit/(1024*1024)),
```
Comment (richardsliu) [inline pkg/gpu/nvidia/manager.go:318]: Can we rename this "memoryLimitBytes" to be less ambiguous?

Comment (grac3gao) [reply pkg/gpu/nvidia/manager.go:318]: Done

File: pkg/gpu/nvidia/manager_test.go (hunk 1 above)
Comment (grac3gao) [reply pkg/gpu/nvidia/manager_test.go:172]: Done

File: pkg/gpu/nvidia/manager_test.go (hunk 2 above)
Comment (grac3gao) [reply pkg/gpu/nvidia/manager_test.go:183]: done.

**Review by richardsliu — APPROVED** [review]

**Pull request merged**